pkg/asset/installconfig/azure/client.go

```diff
@@ -152,3 +156,31 @@ func (c *Client) getProvidersClient(ctx context.Context) (azres.ProvidersClient,
 	client.Authorizer = c.ssn.Authorizer
 	return client, nil
 }
+
+// ValidateResourceSkuAvailability checks if the diskType exists for the given region.
+func (c *Client) ValidateResourceSkuAvailability(ctx context.Context, region string, diskType string) (*azsku.ResourceSku, error) {
```

Comment:
don't return the sku since this is a validation function, should return only error.